### PR TITLE
description added

### PR DESCRIPTION
**Pull request by mayaalsamarrai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,22 +1,52 @@
...
+1. General API request that allows saving the entire shopping cart information on the database;
+1. Specific API requests that allow call specific operations, for example delete, edit, choose payment method, choose delivery type, etc.
+
+For more details about the available cart-module API, please follow the link bellow:
+
+https://admin-demo.virtocommerce.com/docs/ui/index#/Shopping%2520cart%2520module
+
+The cart-module functionality can be extended and customized based on the specific business needs. In order to extend the cart module functionality, use the carte module template that can be accessed by following the link below:
+
+https://marketplace.visualstudio.com/items?itemName=VirtoCommerce2xModuleProjectTemplate.VirtoCommerce2xModule
```
Comment (eMazeika) [inline README.md]: current link is https://marketplace.visualstudio.com/items?itemName=Virto-Commerce.VirtoCommerceModuleTemplates

File: README.md [hunk 2]
```diff
@@ -1,22 +1,52 @@
...
+## Installation
 Installing the module:
 * Automatically: in VC Manager go to Configuration -> Modules -> Shopping cart module -> Install
 * Manually: download module zip package from https://github.com/VirtoCommerce/vc-module-cart/releases. In VC Manager go to Configuration -> Modules -> Advanced -> upload module package -> Install.
 
-# Available resources
+## Available resources
 * Module related service implementations as a <a href="https://www.nuget.org/packages/VirtoCommerce.CartModule.Data" target="_blank">NuGet package</a>
 * API client as a <a href="https://www.nuget.org/packages/VirtoCommerce.CartModule.Client" target="_blank">NuGet package</a>
 * API client documentation http://admin-demo.virtocommerce.com/docs/ui/index#!/Shopping_cart_module
```
Comment (eMazeika) [inline README.md]: correct link: https://admin-demo.virtocommerce.com/docs/ui/index#/Shopping%2520cart%2520module

File: README.md [hunk 3]
```diff
@@ -1,22 +1,52 @@
...
+
+1. Supports multiple carts
+1. Wishlist
+1. Grouping multiple carts to one order
+1. Anonymous carts
+1. Stock reservation
+1. Multiple payments methods
+1. Create new cart from orders history
+
+The main purpose of the cart module is to save and store the shopping cart data and changes on VC Manager side.
```
Comment (eMazeika) [inline README.md]: The main purpose of the Cart module is to implement customer shopping cart management in VC eCommerce solution. It encapsulates data persistence, management services and exposes REST API endpoints.

File: README.md [hunk 4]
```diff
@@ -1,22 +1,52 @@
...
+The cart module is connected with the shopping cart via two types of API requests:
+
+1. General API request that allows saving the entire shopping cart information on the database;
+1. Specific API requests that allow call specific operations, for example delete, edit, choose payment method, choose delivery type, etc.
+
+For more details about the available cart-module API, please follow the link bellow:
+
+https://admin-demo.virtocommerce.com/docs/ui/index#/Shopping%2520cart%2520module
+
+The cart-module functionality can be extended and customized based on the specific business needs. In order to extend the cart module functionality, use the carte module template that can be accessed by following the link below:
```
Comment (eMazeika) [inline README.md]: change
"use the carte module template" to
"use "Virto Commerce 2.x Cart and Order Module extension" template"

File: README.md [hunk 5]
```diff
@@ -1,22 +1,52 @@
...
+1. Wishlist
+1. Grouping multiple carts to one order
+1. Anonymous carts
+1. Stock reservation
+1. Multiple payments methods
+1. Create new cart from orders history
+
+The main purpose of the cart module is to save and store the shopping cart data and changes on VC Manager side.
+
+The cart module is connected with the shopping cart via two types of API requests:
```
Comment (eMazeika) [inline README.md]: Not sure what the idea was, but there is no "two types of API requests".

File: README.md (hunk 1 above)
Comment (mayaalsamarrai) [reply README.md]: Got it !OK Actually I nserted the link that Maksim gave me, looks like it was incorrect

File: README.md (hunk 2 above)
Comment (mayaalsamarrai) [reply README.md]: OK

File: README.md (hunk 5 above)
Comment (mayaalsamarrai) [reply README.md]: I received this info from Eugene Tatarintsev, may be I did not get it well) How would you recommend to refrase it?

**Comment by mvktsk** [comment]
@mayaalsamarrai general comment. In the article used 3 different terms: "VirtoCommerce.Cart module", "Shopping cart module", "Cart module". Plese chosee 1 for article.

File: README.md [hunk 6]
```diff
@@ -1,22 +1,52 @@
...
+VirtoCommerce.Cart module represents shopping cart management system. This module doesn't have any UI in VC Manager.
 
-# Installation
+Shopping cart module manages customers accumulated list of items, calculates a total for the order, including shipping and handling (i.e., postage and packing) charges and the associated taxes.
+
+![Cart Module](docs/media/screen-cart-module.png)
+
+![Cart Module Info](docs/media/screen-cart-module-info.png)
+
+## Features
```
Comment (mvktsk) [inline README.md]: Please add a short description of each features, 1-2 paragraphs, not more for feature.

File: README.md [hunk 7]
```diff
@@ -1,22 +1,52 @@
...
+1. Stock reservation
+1. Multiple payments methods
+1. Create new cart from orders history
+
+The main purpose of the Cart module is to implement customer shopping cart management in VC eCommerce solution. It encapsulates data persistence, management services and exposes REST API endpoints.
+
+The cart module is connected with the shopping cart API requests:
+
+1. General API request that allows saving the entire shopping cart information on the database;
+1. Specific API requests that allow call specific operations, for example delete, edit, choose payment method, choose delivery type, etc.
```
Comment (mvktsk) [inline README.md]: @eMazeika @mayaalsamarrai For discuss. I think it would be nice to add a list of API functions and their brief description

Comment (eMazeika) [reply README.md]: link to live swagger would be better, imho. Can add descriptions in api level, if missing.

File: README.md (hunk 5 above)
Comment (eMazeika) [reply README.md]: @tatarincev , could you clarify on "two types of API requests"?

File: README.md (hunk 6 above)
Comment (mayaalsamarrai) [reply README.md]: OK

File: README.md (hunk 7 above)
Comment (mayaalsamarrai) [reply README.md]: Guys, what's to technical stuff, like description in api level, etc, I would ask you to help me with that. Please provide the details and I will add it to the document

File: README.md [hunk 8]
```diff
@@ -1,22 +1,59 @@
...
-# Installation
+VirtoCommerce.Cart module manages customers accumulated list of items, calculates a total for the order, including shipping and handling (i.e., postage and packing) charges and the associated taxes.
+
+![Cart Module](docs/media/screen-cart-module.png)
+
+![Cart Module Info](docs/media/screen-cart-module-info.png)
+
+## Features
+
+1. Supports multiple carts- if the user is using more than one cart at the same time, all of the carts will be supported by VirtoCommerce.Cart module.
```
Comment (mvktsk) [inline README.md]: add space before -

File: README.md [hunk 9]
```diff
@@ -1,22 +1,59 @@
...
+
+1. Wishlist-
+the user can add the desired products to the  wishlist , which will be saved in the cart. The user can later use the products added to the wishlist to make an order. The wishlist details will be saved by the VirtoCommerce.Cart module.
+1. Grouping multiple carts to one order-
+If the user is using more than one cart, the final order will be created by grouping the details from all the carts filled out by user.
```
Comment (mvktsk) [inline README.md]: should be on same line with  "Grouping multiple carts to one order"
If  should be lower case

File: README.md [hunk 10]
```diff
@@ -1,22 +1,59 @@
...
+
+![Cart Module](docs/media/screen-cart-module.png)
+
+![Cart Module Info](docs/media/screen-cart-module-info.png)
+
+## Features
+
+1. Supports multiple carts- if the user is using more than one cart at the same time, all of the carts will be supported by VirtoCommerce.Cart module.
+
+1. Wishlist-
```
Comment (mvktsk) [inline README.md]: add space before -

File: README.md [hunk 11]
```diff
@@ -1,22 +1,59 @@
...
+the user can add the desired products to the  wishlist , which will be saved in the cart. The user can later use the products added to the wishlist to make an order. The wishlist details will be saved by the VirtoCommerce.Cart module.
+1. Grouping multiple carts to one order-
+If the user is using more than one cart, the final order will be created by grouping the details from all the carts filled out by user.
+1. Anonymous carts-
```
Comment (mvktsk) [inline README.md]: add space before -

File: README.md [hunk 12]
```diff
@@ -1,22 +1,59 @@
...
+the user is allowed to use an anonymous account, not necessarily to login under his own account. Such carts information will also be saved on the database by the cart module.
+1. Stock reservation-
```
Comment (mvktsk) [inline README.md]: add space before -

File: README.md [hunk 13]
```diff
@@ -1,22 +1,59 @@
...
+the selected products will be reserved in stock after adding them to the cart.
+1. Multiple payments methods-
```
Comment (mvktsk) [inline README.md]: add space before -

File: README.md [hunk 14]
```diff
@@ -1,22 +1,59 @@
...
+the user can choose the payment methods he wants before checkout.
+1. Create new cart from orders history-
```
Comment (mvktsk) [inline README.md]: add space before -

File: README.md [hunk 15]
```diff
@@ -1,22 +1,59 @@
...
+## Features
+
+1. Supports multiple carts- if the user is using more than one cart at the same time, all of the carts will be supported by VirtoCommerce.Cart module.
+
+1. Wishlist-
+the user can add the desired products to the  wishlist , which will be saved in the cart. The user can later use the products added to the wishlist to make an order. The wishlist details will be saved by the VirtoCommerce.Cart module.
+1. Grouping multiple carts to one order-
+If the user is using more than one cart, the final order will be created by grouping the details from all the carts filled out by user.
+1. Anonymous carts-
+the user is allowed to use an anonymous account, not necessarily to login under his own account. Such carts information will also be saved on the database by the cart module.
```
Comment (mvktsk) [inline README.md]: should be on same line with "Anonymous carts"

File: README.md [hunk 16]
```diff
@@ -1,22 +1,59 @@
...
+1. Stock reservation-
+the selected products will be reserved in stock after adding them to the cart.
```
Comment (mvktsk) [inline README.md]: should be on same line with "Stock reservation"

File: README.md [hunk 17]
```diff
@@ -1,22 +1,59 @@
...
+1. Multiple payments methods-
+the user can choose the payment methods he wants before checkout.
```
Comment (mvktsk) [inline README.md]: should be on same line with "Multiple payments methods"

File: README.md [hunk 18]
```diff
@@ -1,22 +1,59 @@
...
+1. Create new cart from orders history-
+the user can add products to a new cart using the previously completed orders (orders history).
```
Comment (mvktsk) [inline README.md]: should be on same line with "Create new cart from orders history"

File: README.md (hunk 7 above)
Comment (mvktsk) [reply README.md]: To describe the API, you should add "Documentation comments" https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments
You can find examples in /VirtoCommerce.CartModule.Web/Controllers/Api/CartModuleController.cs
Its look like 

/// <summary>
/// Get shopping cart by id
/// </summary>

Comment (mayaalsamarrai) [reply README.md]: A task in Jira created

File: README.md [hunk 19]
```diff
@@ -1,22 +1,53 @@
...
+
+![Cart Module](docs/media/screen-cart-module.png)
+
+![Cart Module Info](docs/media/screen-cart-module-info.png)
+
+## Features
+
+1. Supports multiple carts - if the user is using more than one cart at the same time, all of the carts will be supported by VirtoCommerce.Cart module.
+
+1. Wishlist - the user can add the desired products to the  wishlist , which will be saved in the cart. The user can later use the products added to the wishlist to make an order. The wishlist details will be saved by the VirtoCommerce.Cart module.
```
Comment (eMazeika) [inline README.md]: The feature is called "Named lists", while Wishlist is just one example of the names. "Gifts" could be another list name example.

File: README.md [hunk 20]
```diff
@@ -1,22 +1,53 @@
...
+1. Grouping multiple carts to one order - if the user is using more than one cart, the final order will be created by grouping the details from all the carts filled out by user.
+1. Anonymous carts - the user is allowed to use an anonymous account, not necessarily to login under his own account. Such carts information will also be saved on the database by the cart module.
```
Comment (eMazeika) [inline README.md]: Anonymous carts - user can put products to cart and submit order anonymously, without creating an account.

File: README.md [hunk 21]
```diff
@@ -1,22 +1,53 @@
...
+https://admin-demo.virtocommerce.com/docs/ui/index#/Shopping%2520cart%2520module
+
+The VirtoCommerce.Cart module functionality can be extended and customized based on the specific business needs. In order to extend the cart module functionality, use "Virto Commerce 2.x Cart and Order Module extension" template that can be accessed by following the link below:
+
+https://marketplace.visualstudio.com/items?itemName=Virto-Commerce.VirtoCommerceModuleTemplates
+
+
+## Installation
 Installing the module:
 * Automatically: in VC Manager go to Configuration -> Modules -> Shopping cart module -> Install
```
Comment (eMazeika) [inline README.md]: Automatically: in VC Manager go to More -> Modules -> Shopping cart module -> Install

File: README.md [hunk 22]
```diff
@@ -1,22 +1,53 @@
...
 * Manually: download module zip package from https://github.com/VirtoCommerce/vc-module-cart/releases. In VC Manager go to Configuration -> Modules -> Advanced -> upload module package -> Install.
```
Comment (eMazeika) [inline README.md]: change Configuration  to More

File: README.md [hunk 23]
```diff
@@ -1,22 +1,53 @@
...
 
-# Available resources
+## Available resources
 * Module related service implementations as a <a href="https://www.nuget.org/packages/VirtoCommerce.CartModule.Data" target="_blank">NuGet package</a>
 * API client as a <a href="https://www.nuget.org/packages/VirtoCommerce.CartModule.Client" target="_blank">NuGet package</a>
```
Comment (eMazeika) [inline README.md]: remove this point: we no longer support/provide API client packages on nuget

**Pull request merged**